txnkv/transaction/txn.go

```diff
@@ -162,6 +164,9 @@ type KVTxn struct {
 	aggressiveLockingDirty   atomic.Bool
 
 	forUpdateTSChecks map[string]uint64
+
+	isPipelined    bool
+	pipelinedMemDB *unionstore.PipelinedMemDB
```

Comment:
Is there any case that we may need both `us` and `pipelinedMemDB` in a same transaction? I thought we should make a new interface and make the original unionstore and PipelinedMemDB 2 implementations of the new interface. So only one field is needed here.

Reply:
> Is there any case that we may need both us and pipelinedMemDB in a same transaction?

No.

> make a new interface and make the original unionstore and PipelinedMemDB 2 implementations of the new interface.

There are 2 concerns:

* TiDB strongly depends on `GetUnionStore()` and `GetUnionStore().GetMemBuffer()`, with an interface, unionstore cannot be accessed directly.
* The `Get` function may introduce a tier parameter.

I'll try this abstraction.

Reply:
I made an interface for `MemDB` and `PipelinedMemDB`, it's named `MemBuffer`. I think we can reuse the snapshot read code in `KVUnionStore`

See the changes in `union_store.go`.

The only change to user interface is `Get` now accept a context, this may break compile for projects which import client-go.